scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -37,6 +43,16 @@ def process_response(self, request, response, spider):
             if content_encoding:
                 encoding = content_encoding.pop()
                 decoded_body = self._decode(response.body, encoding.lower())
+                if self._max_size and len(decoded_body) > self._max_size:
```

Comment:
It'd be also nice to figure out if we can support this option for deflate and brotli

Reply:
Indeed, I was toying around with pushing this down to the gunzip method, but decided against it because I anyway want to keep a highlevel check to make sure this covers all decompressing algorithms brotly/deflate 

Happy to complement by adding an optional argument to gunzip(data, max_size=0) and abort sooner,

Reply:
+1 to also support brotli/deflate, I couldn't find a trivial way to accomplish that, other than rolling our own brotli decompression function (similar to what we do for gunzip [which tbh I don't fully understand the rationale behind]), 

Afraid this PR becomes too big if we want to implement this for brotli/deflate, but happy to follow your advice here

Thanks!

Reply:
Right, it makes sense to support it both at gunzip level and on a lavel higher - at least it'd allow this option to work for brotli/deflate, even though performance won't be there yet.